packages/api-server/api_server/test_sio_auth.py

```diff
@@ -39,7 +39,9 @@ def try_connect(self, token: Optional[str] = None) -> bool:
                     client.connect(self.base_url, auth=auth)
                     return True
                 except socketio.exceptions.ConnectionError as e:
-                    if str(e) == "Connection refused by the server" and tries < 10:
+                    # Check specifically for connection refused "Errno 111"
+                    # Likely a race condition, wait for server to start
+                    if "Errno 111" in str(e) and tries < 10:
```

Comment:
What if we reverse this check? If we know the error caused by bad authentication, we can check for that instead, and retry for all other errors.

Reply:
the error message returned when authentication fails is of type ConnectionError with string "One or more namespaces failed to connect" which is not very descriptive. The source is [here](https://github.com/miguelgrinberg/python-socketio/blob/main/src/socketio/client.py#L347-L348), its not specific to authentication errors. Its not obvious to me how to explicitly target authentication errors.

separately, i'm thinking we can replace the "tries" variable with https://github.com/miguelgrinberg/python-socketio/blob/main/src/socketio/client.py#L44-L46, Let me know what you think?

Reply:
That at least tells us if it is because the socket fails to connect or the socketio protocol fails to connect. We want to retry only when the socket fails, so we should not be using `reconnection_attempt`.

To be fair, the current code doesn't detect authentication error as well, retries when there is a socket error, and just return false when it is any other error (which may not be authentication error, or even socketio errors). If it is easier to detect any socketio error, we can instead return false when it happens and retry for all other errors.

Reply:
ef1fcfc535ef39bcee3ad712721085b0464ce09a should (hackily, unfortunately) test for this using a string comparison.